runtime/interpreter/value.go

```diff
@@ -18043,7 +18059,7 @@ func (v *DictionaryValue) ForEachKey(
 	}
 
 	iterate := func() {
-		err := v.dictionary.IterateKeys(
+		err := v.dictionary.IterateReadOnlyKeys(
```

Comment:
Are we sure that the `procedure` passed to `DictionaryValue.ForEachKey` does not mutate? I'd be conservative and switch to a read-only iterator after we ensured it. It might also be good to indicate that in the name then

Reply:
> Are we sure that the procedure passed to DictionaryValue.ForEachKey does not mutate?

Hmm, do you mean the procedure might mutate map value by iterated key?  

Using a non-readonly vs readonly iterator probably wouldn't make a difference here because mutation through iterator is done by callback in returned map value.  Since procedure only processes map key, mutating map value is not possible by callback.  Actually, atree only has readonly iterators for map keys for the same reason.

Thoughts?

Reply:
> [...] because mutation through iterator is done by callback in returned map value

Not quite sure what that means. Do you maybe have an example?

> mutating map value is not possible by callback

The  callback is free to do anything, including removing keys, inserting new keys, mutating values in the dictionary, etc.

I guess your point is still that mutable vs read-only iterator doesn't matter?